Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+
+        # only pass through things known to pandas, e.g.
+        # Table(..., weights=1) passes weights to from_numpy, but would error
+        # when passing to pandas upstream
+        super().__init__(**{k: v for k, v in kwargs.items() if k in self._KNOWN_PANDAS_KWARGS})
+
+        self.name = getattr(self, 'name', kwargs.get("name", "untitled"))
+        self.attributes = getattr(self, 'attributes', kwargs.get("attributes", {}))
+        self.__file__ = getattr(self, '__file__', kwargs.get("__file__"))
+
+        # we need to filter the domain to only include the columns present in the table
+        # but we still need to allow constructing an empty table (with no domain)
+        # also, we only set the domain if it has changed (==number of variables),
+        # so in those cases, id(domain_before) == id(domain_after)
+        if hasattr(self, 'domain') and self.domain is not None:
+            new_domain = Domain(
+                [c for c in self.domain.attributes if c in self.columns],
+                [c for c in self.domain.class_vars if c in self.columns],
+                [c for c in self.domain.metas if c in self.columns]
+            )
+            if len(new_domain.variables) + len(new_domain.metas) != \
+                            len(self.domain.variables) + len(self.domain.metas):
+                self.domain = new_domain
+        else:
+            self.domain = None
+
+        # only set the weights if they aren't set already
+        if self._WEIGHTS_COLUMN not in self.columns:
+            self.set_weights(1)
+
+    @classmethod
+    def from_domain(cls, domain):
+        """Construct a new Table for the given Domain.
+
+        Parameters
+        ----------
+        domain : Domain
+            The domain passed to the new Table object.
+
+        Returns
+        -------
+        TableBase
+            The new Table with a Domain assigned.
+        """
+        result = cls(columns=domain.attributes + domain.class_vars + domain.metas)
+        result.domain = domain
+        return result
+
+    @classmethod
+    @deprecated("t.iloc[row_indices].copy()")
+    def from_table_rows(cls, source, row_indices):
+        """Construct a new Table as a copy of the selected rows (position indexing).
+
+        Parameters
+        ----------
+        source : TableBase
+            The source Table.
+        row_indices
+            The locations of the selected rows in any of the standard formats.
+        Returns
+        -------
+        TableBase
+            A subset of the original Table as a copy.
+        """
+        # don't just plain copy here: in case of subclasses of Table, a plain table is passed
+        # through the constructor and from_table to here, and expects to be converted
+        # into a proper subclass type
+        result = cls(data=source.iloc[row_indices]).copy()
+        result._transfer_properties(source, transfer_domain=True)  # because we manually copy data, not the whole table
```

Comment:
`_metadata` aren't already copied on `copy()`. They [should be](http://stackoverflow.com/questions/28041762/pandas-metadata-of-dataframe-persistence-error).